lib/hotel.rb

```diff
@@ -0,0 +1,57 @@
+require_relative 'room'
+require_relative 'reservation'
+
+module Hotel
+
+  class Hotel
+    include Reservable
+
+    NUM_ROOMS = 20
+
+    attr_reader :rooms
+
+    def initialize(num_rooms = NUM_ROOMS)
+      # check input
+      check_room_num(num_rooms)
+
+      @rooms = {}
+
+      # loop through num_rooms and add rooms to hash
+      (1..num_rooms).each do |num|
+        @rooms[num] = ::Hotel::Room.new(num)
+      end
+
+    end
+
+    def reserve(start_date, end_date, room)
```

Comment:
Just an observation, a user would need to identify a room object first before trying to reserve it.  Why not just have them provide a room number and have this method first find the room.  

That way the user doesn't have to know about Room objects or the Room class.